tests/test_debug_utils_redact.py

```
@@ -232,24 +232,69 @@ def test_exported_in_all(self) -> None:
+      * Delete a name from the constant AND from the transcription -- which is exactly
+        what the pin's own failure message instructs a developer to do -- and the pin
+        is neutralised by construction. Nine such edits were run; the pin appeared in
+        none of the failure lists, and four of them (mac_address, mobile_id,
+        serial_number, transaction_id) left the WHOLE suite green.
+
+    So the assertions below drive redact_identity itself. Making a retired key green
+    again now requires DELETING a line that says that key must be masked, and watching
+    a canary credential appear in the output -- not syncing a bookkeeping literal.
+    """
+
+    _SECRET = "CANARY-CREDENTIAL-VALUE"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "_SECRET"

(S105)